src/tokens/generateHelpers.ts

```diff
@@ -58,6 +61,35 @@ type Helpers<
 					// Otherwise, we get infinite depth issue.
 			  }) => Path extends string ? ObjectPathValue<Theme[K], Path> : never
 		: () => (props: { theme: Theme }) => Theme[K];
+} & (BreakpointsKey extends keyof Theme
+	? {
+			mediaQuery: {
+				min: (
+					size: keyof Theme[BreakpointsKey],
+				) => (props: { theme: Theme }) => string;
+				max: (
+					size: keyof Theme[BreakpointsKey],
+				) => (props: { theme: Theme }) => string;
+			};
+	  }
+	: Record<string, never>);
+
+const constructMediaQuery = <
+	BreakpointsKey extends string,
```

Comment:
Should this one use the same `BreakpointsKey extends keyof Theme`?

Reply:
This is a bit different use-case and couldn't do `extends keyof Theme` here. As the breakpoints are optional, we can't guarantee that breakpoints will be part of Theme – it may not be part of keyof theme.

Reply:
thanks for clarifying <3